test/04-verifying/35-diffheaders.js

```diff
@@ -152,3 +143,41 @@ test('test diffheaders return values adding From header field', async t => {
   }
 });
 
+test('Memory Leak Test For DiffHeaders Variable', async t => {
+  try {
+    return iterate.async(async () => {
+      var opendkim = new OpenDKIM();
+      opendkim.query_method('DKIM_QUERY_FILE');
+      opendkim.query_info('../fixtures/testkeys');
+      await opendkim.verify({id: undefined});
+      var header = 'From: Murray S. Kucherawy <yo@mismo.com>';
+      await opendkim.header({
+        header: header,
+        length: header.length
+      });
+      for (var i = 0; i < 3; i++) {
+        header = 'Received: received data ' + i;
+        await opendkim.header({
+          header: header,
+          length: header.length
+        });
+      }
+      await opendkim.chunk({
+        message: messages.good_z,
+        length: messages.good_z.length
+      });
+      await opendkim.chunk_end();
+      opendkim.diffheaders({
+        maxcost: 10
+      });
+    }, {iterations: 10, gcollections: 6})
```

Comment:
Keep itearations small. If not the test will take ages to complete.
Nevertheless I've tested with 100 iteration and still no leaks.